src/agent_factory/agent_generator.py

```diff
@@ -48,7 +50,12 @@ async def generate_target_agent(
         timeout: The timeout for the request in seconds (default: 600).
     """
     with tracer.start_as_current_span("generate_target_agent") as span:
-        trace_file = f"0x{trace.format_trace_id(span.get_span_context().trace_id)}.jsonl"
+        trace_id = trace.format_trace_id(span.get_span_context().trace_id)
+        spans_dump_file_path = TRACES_DIR / f"0x{trace_id}.jsonl"
+        # The same trace_id is used as the folder name when saving agent artifacts (on local/MinIO/S3)
+        output_dir = output_dir if output_dir else trace_id
```

Comment:
Just for transparency, could you add in the PR description that now the output_dir will be using the `trace_id` instead of `request.id` as a name? Was there a specific reason for this change btw?